relay-sampling/src/dsc.rs

```diff
@@ -111,8 +111,7 @@ impl DynamicSamplingContext {
 
     /// Compute the effective sampling rate based on the random "diceroll" and the sample rate from
     /// the matching rule.
-    pub fn adjusted_sample_rate(&self, rule_sample_rate: f64) -> f64 {
-        let client_sample_rate = self.sample_rate.unwrap_or(1.0);
+    pub fn adjusted_sample_rate(client_sample_rate: f64, rule_sample_rate: f64) -> f64 {
```

Comment:
i found it ugly to pass in the entire sampling context just to get the `client_sample_rate` here: https://github.com/getsentry/relay/pull/2514/files#diff-c36d16143aa181bfcd1fca3ca1fba4f23ee13c549c08083ba67895bbb815436fR286